addons/mail/static/src/messaging/component/notification_group/notification_group.js

```diff
@@ -0,0 +1,84 @@
+odoo.define('mail.messaging.component.NotificationGroup', function (require) {
+'use strict';
+
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+class NotificationGroup extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const group = this.env.entities.NotificationGroup.get(props.notificationGroupLocalId);
+            return {
+                group: group ? group.__state : undefined,
+            };
+        });
+        /**
+         * Reference of the "mark as read" button. Useful to disable the
+         * top-level click handler when clicking on this specific button.
+         */
+        this._markAsReadRef = useRef('markAsRead');
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {mail.messaging.entity.Notification}
+     */
+    get group() {
+        return this.env.entities.NotificationGroup.get(this.props.notificationGroupLocalId);
+    }
+
+    /**
+     * @returns {string|undefined}
+     */
+    image() {
+        if (this.group.notification_type === 'email') {
+            return '/mail/static/src/img/smiley/mailfailure.jpg';
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClick(ev) {
+        if (ev.target === this._markAsReadRef.el) {
+            // handled in `_onClickFailureDiscard`
+            return;
+        }
```

Comment:
The intent is good, but the current solution is not satisfying: it assumes ref has no children and is not a component (if it becomes a component, we have to change this LOC, which kinda sucks... even so if this component also has children and/or sub-components, this becomes annoying to check).

Reply:
Changed to `contains` instead of equality.

As for `.el` or `.comp` there is nothing we can do since this is how `useRef` works, we already questioned/reported the annoyance of it, but this shouldn't prevent us from using it.